src/Integration.Vsix.UnitTests/SonarLintTagger/AccumulatingIssueConsumerTests.cs

```diff
@@ -0,0 +1,182 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.VisualStudio.Text;
+using Moq;
+using SonarLint.VisualStudio.Core.Analysis;
+using SonarLint.VisualStudio.Integration.Vsix;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests.SonarLintTagger
+{
+    [TestClass]
+    public class AccumulatingIssueConsumerTests
+    {
+        private static readonly IAnalysisIssue ValidIssue = CreateIssue(startLine: 1, endline: 1);
+        private static readonly ITextSnapshot ValidTextSnapshot = CreateSnapshot(lineCount: 10);
+        private const string ValidFilePath = "c:\\myfile.txt";
+
+        [TestMethod]
+        public void Ctor_InvalidArgs_Throws()
+        {
+            AccumulatingIssueConsumer.OnIssuesChanged validCallback = _ => { };
+
+            Action act = () => new AccumulatingIssueConsumer(null, ValidFilePath, validCallback);
+            act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("analysisSnapshot");
+
+            act = () => new AccumulatingIssueConsumer(ValidTextSnapshot, null, validCallback);
+            act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("analysisFilePath");
+
+            act = () => new AccumulatingIssueConsumer(ValidTextSnapshot, ValidFilePath, null);
+            act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("onIssuesChangedCallback");
+        }
+
+        [TestMethod]
+        public void Accept_WrongFile_CallbackIsNotCalled()
+        {
+            var callbackSpy = new OnIssuesChangedCallbackSpy();
+            var issues = new IAnalysisIssue[] { ValidIssue };
+
+            var testSubject = new AccumulatingIssueConsumer(ValidTextSnapshot, "c:\\file1.txt", callbackSpy.Callback);
+
+            using (new AssertIgnoreScope())
+            {
+                testSubject.Accept("wrong file", issues);
+            }
+
+            callbackSpy.CallCount.Should().Be(0);
+        }
+
+        [TestMethod]
+        [DataRow(-1, 10, false)] // start line < 1
+        [DataRow(0, 0, false)] // file-level issue, can't be mapped to snapshot
+        [DataRow(1, 1, true)] // starts in first line of snapshot
+        [DataRow(9, 10, true)] // in snapshot
+        [DataRow(10, 10, true)] // end is in last line of snapshot
+        [DataRow(10, 11, false)] // end is outside snapshot
+        [DataRow(11, 11, false)] // ditto
```

Comment:
It's a bit dangerous IMO to leave the "ditto" comment, if someone were to change the order of the cases, or add something in between, it would change the meaning. Maybe just copy-paste the comment.